src/ResourceManagement/EventHub/EventHub.Tests/TestHelper/EventHubManagementHelper.cs

```diff
@@ -0,0 +1,151 @@
+        internal const string ConsumerGroupPrefix = "sdk-ConsumerGroup-";
+
+
+        public static EventHubManagementClient GetEventHubManagementClient(MockContext context, RecordedDelegatingHandler handler)
+        {
+            if (handler != null)
+            {
+                handler.IsPassThrough = true;
+                EventHubManagementClient nhManagementClient = context.GetServiceClient<EventHubManagementClient>(handlers: handler);
+                return nhManagementClient;
+            }
+
+            return null;
+        }
+
+        public static ResourceManagementClient GetResourceManagementClient(MockContext context, RecordedDelegatingHandler handler)
+        {
+            if (handler != null)
+            {
+                handler.IsPassThrough = true;
+                ResourceManagementClient rManagementClient = context.GetServiceClient<ResourceManagementClient>(handlers: handler);
+                return rManagementClient;
+            }
+
+            return null;
+        }
+
+        private static void ThrowIfTrue(bool condition, string message)
+        {
+            if (condition)
+            {
+                throw new Exception(message);
+            }
+        }
+
+        public static string TryGetResourceGroup(this ResourceManagementClient resourceManagementClient, string location)
+        {
+            var resourceGroup =
+                resourceManagementClient.ResourceGroups
+                    .List().Where(group => string.IsNullOrWhiteSpace(location) || group.Location.Equals(location, StringComparison.OrdinalIgnoreCase))
+                    .FirstOrDefault(group => group.Name.Contains(""));
+
+            return resourceGroup != null
+
+                ? resourceGroup.Name
+                : string.Empty;
+        }
+        
+        public static void TryRegisterResourceGroup(this ResourceManagementClient resourceManagementClient, string location, string resourceGroupName)
+        {
+            resourceManagementClient.ResourceGroups.CreateOrUpdate(resourceGroupName, new ResourceGroup(location));
+        }
+        
+        public static void TryCreateNamespace(
+            this EventHubManagementClient client,
+            string resourceGroupName,
+            string namespaceName,
+            string location,
+            string scaleUnit = null)
+        {
+            var namespaceParameter = new NamespaceCreateOrUpdateParameters()
+            {
+                Location = location
+            };            
+
+            client.Namespaces.CreateOrUpdate(
+                resourceGroupName,
+                namespaceName, namespaceParameter
+               );
+
+            var response = client.Namespaces.Get(resourceGroupName, namespaceName);
+            ThrowIfTrue(!response.Name.Equals(namespaceName), string.Format("Namespace name is not equal to {0}", namespaceName));
+        }
+
+        public static string GenerateRandomKey()
```

Comment:
Please use the TestUtilities for this, as in here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1128/

This will store random values in the test recording automatically.